q0018/q0018.py

```diff
@@ -0,0 +1,32 @@
+class Solution:
+    def fourSum(self, nums: List[int], target: int) -> List[List[int]]:
+        nums.sort()
+        res = []
+        ele = []
+        # k is the number of elements left to add in each quad
+        # start is the starting place for the quad
+        def helper(k, start, target): 
+            # if there are still more than two elements left, then tackle it down to two sum
+            if k > 2:
+                for i in range(start, len(nums) - k + 1):
+                    if i > start and nums[i] == nums[i - 1]:
+                        continue
+                    ele.append(nums[i])
+                    helper(k - 1, i + 1, target - nums[i])
+                    ele.pop()
+                return
+            # this part will be reached when there are two elements left for the quad element
+            # this part is the same as two sum problem
+            l , r = start, len(nums) - 1

```

Comment:
Would recommend to extract this into its own function, but I think that could be done later as a follow up.